packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,244 @@
+                }
+                return { success: true };
+            });
+
+            await expect(funcToRetry).retry().and.have.property('success').and.be.true;
+
+            expect(getCallCount()).to.equal(3);
+        });
+
+        it('.sealed', async () => {
+            const sealedObject = Object.seal({ prop1: 'value1', prop2: 'value2' });
+
+            await expect(() => sealedObject).retry().to.be.sealed;
+        });
+
+        it('.not.sealed', async () => {
+            const notSealedObject = { prop1: 'value1', prop2: 'value2' };
+
+            try {
+                await expect(() => notSealedObject).retry({ retries: 3 }).to.be.sealed;
+            } catch (error: unknown) {
+                expect((error as Error).message).to.include('3 retries');
+            }
+        });
+    });
+
+    describe('should work with various assertion methods:', () => {
+        it('.satisfy()', async () => {
+            const { funcToRetry, getCallCount } = withCallCount((callCount) => ({
+                callCount,
+            }));
+
+            await expect(funcToRetry)
+                .retry()
+                .to.satisfy((obj: { callCount: number }) => obj.callCount > 3);
+
+            expect(getCallCount()).to.equal(4);
+        });
+
+        it('.have.property(), and.be.above()', async () => {
+            const { funcToRetry, getCallCount } = withCallCount((callCount) =>
+                callCount < 4 ? { notValue: 2 } : { value: callCount }
+            );
+
+            await expect(funcToRetry).retry().have.property('value').and.be.above(4);
+
+            expect(getCallCount()).to.equal(5);
+        });
+
+        it('.to.increase(), .by()', async () => {
+            const myObj = { val: 1 };
+            const { funcToRetry, getCallCount } = withCallCount(() => {
+                myObj.val += 2;
+            });
+
+            await expect(funcToRetry).retry().to.increase(myObj, 'val').by(2);
+
+            expect(getCallCount()).to.equal(1);
+        });
+
+        it('.oneOf()', async () => {
+            const { funcToRetry, getCallCount } = withCallCount((callCount) => callCount);
+
+            await expect(funcToRetry).retry().to.not.be.oneOf([1, 2, 3]);
+            expect(getCallCount()).to.equal(4);
+        });
+
+        it('.change()', async () => {
+            const myObj = { dots: '', comas: '' };
+
+            const { funcToRetry, getCallCount } = withCallCount(() => {
+                myObj.dots += '.';
+            });
+
+            await expect(funcToRetry).retry().to.change(myObj, 'dots');
+            await expect(funcToRetry).retry().to.not.change(myObj, 'comas');
+
+            expect(getCallCount()).to.equal(2);
+        });
+    });
+});
+
+const withCallCount = (func: (callCount: number) => unknown) => {
+    let callCount = 0;
+
+    const wrapperFunc = () => {
+        callCount++;
+        return func(callCount);
+    };
+
+    return {
+        funcToRetry: wrapperFunc,
```

Comment:
it is not a good name - even though it is local to this file, `withCallCount` sounds like generic helper which has nothing to do with `retrying` something.

Reply:
@yurii-ve fixed